src/app/server-cluster/testing/EmptyProvider.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/server-cluster/testing/EmptyProvider.h>
+
+namespace chip {
+namespace Test {
+
+using Protocols::InteractionModel::Status;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+
+CHIP_ERROR EmptyProvider::Shutdown()
+{
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR EmptyProvider::Endpoints(ListBuilder<app::DataModel::EndpointEntry> & builder)
+{
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR EmptyProvider::SemanticTags(EndpointId endpointId, ListBuilder<SemanticTag> & builder)
+{
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR EmptyProvider::DeviceTypes(EndpointId endpointId, ListBuilder<app::DataModel::DeviceTypeEntry> & builder)
+{
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR EmptyProvider::ClientClusters(EndpointId endpointId, ListBuilder<ClusterId> & builder)
+{
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR EmptyProvider::ServerClusters(EndpointId endpointId, ListBuilder<app::DataModel::ServerClusterEntry> & builder)
+{
+    return CHIP_NO_ERROR;
```

Comment:
I guess the API here is undocumented, so maybe returning success and doing nothing for "invalid endpoint" is OK...  But followup: we should document the API.

Reply:
This was a bare minimum to have "something that is not an invalid object so I can write tests".

I also believe this empty provider is nonsense.... It should probably return invalid endpoint on every query instead.

Reply:
Updated to return invalid endpoint except when listing endpoints (which returns an empty list).